src/v/cluster_link/utils/topic_properties_utils.cc

```diff
@@ -0,0 +1,342 @@
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.retention_local_target_ms,
+          config_value,
+          topic_config.properties.retention_local_target_ms);
+    }
+    if (config_name == kafka::topic_property_remote_read) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.remote_read,
+          config_value,
+          topic_config.properties.shadow_indexing.has_value()
+            ? ::model::is_fetch_enabled(
+                topic_config.properties.shadow_indexing.value())
+            : false);
+    }
+    if (config_name == kafka::topic_property_remote_write) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.remote_write,
+          config_value,
+          topic_config.properties.shadow_indexing.has_value()
+            ? ::model::is_fetch_enabled(
+                topic_config.properties.shadow_indexing.value())
+            : false);
+    }
+    if (config_name == kafka::topic_property_remote_delete) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.remote_delete,
+          config_value,
+          topic_config.properties.remote_delete);
+    }
+    if (config_name == kafka::topic_property_max_message_bytes) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.batch_max_bytes,
+          config_value,
+          topic_config.properties.batch_max_bytes);
+    }
+    if (config_name == kafka::topic_property_segment_ms) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.segment_ms,
+          config_value,
+          topic_config.properties.segment_ms);
+    }
+    if (
+      config_name
+      == kafka::topic_property_initial_retention_local_target_bytes) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.initial_retention_local_target_bytes,
+          config_value,
+          topic_config.properties.initial_retention_local_target_bytes);
+    }
+    if (
+      config_name == kafka::topic_property_initial_retention_local_target_ms) {
+        return parse_and_set(
+          topic_config.tp_ns,
+          update.properties.initial_retention_local_target_ms,
+          config_value,
+          topic_config.properties.initial_retention_local_target_ms);
+    }
+    /// XXX: Not sure if we want to support mirroring schema ID validation
+    // if (
+    //   config::shard_local_cfg().enable_schema_id_validation()
+    //   != pandaproxy::schema_registry::schema_id_validation_mode::none) {
+    //     if (schema_id_validation_config_parser(cfg)) {
+    //     }
+    // }
+    if (
+      std::find(
+        std::begin(kafka::allowlist_topic_noop_confs),
+        std::end(kafka::allowlist_topic_noop_confs),
+        config_name)
+      != std::end(kafka::allowlist_topic_noop_confs)) {
+        return false;
+    }
```

Comment:
this seems pretty unnecessary, since the fallthrough behavior for "no match" is to return false.